src/session.rs

```diff
@@ -173,22 +173,33 @@ impl Ongoing {
         Poll::Ready(rt)
     }
 }
+pub(crate) trait Connector:
+    Sized + Sync + Clone + hyper::client::connect::Connect + Unpin + Send + 'static
```

Comment:
I don't think we need `Sized` or `Unpin` here. We _probably_ don't need `Send + Sync + 'static` either?

Reply:
From what I understand we need Send + Sync + 'static to satisfy bounds on hyper::client::Client implementation of Service and Unpin to impl Future and also for the tokio::spawn call in creating the client, maybe I'm going at it wrong but it works for me only with these bounds.

Reply:
Hmm, you may need them for certain `impl` blocks, but I don't think you need all of them everywhere?